### PR TITLE
[chiptool.py] Add --trace_file, --trace_decode, --node_id, --continue…

### LINKED ISSUE #26303: [chiptool.py] `chiptool.py tests` command missing support for several command line arguments used by Matter TH
### Reproduction steps / Feature

**Steps to reproduce:**
1. Build `chip-tool` and `chip-all-clusters-app` in SDK Repo
2. Run `chip-all-clusters-app`
3. Perform "paring" using `chiptool.py` from same ethernet
    
    ```sh
    ./scripts/tests/yaml/chiptool.py pairing onnetwork-long 0x12344321 20202021 3840
    ```
    
4. Attempt to run a test case Matter TH:
    
    ```sh
    ./scripts/tests/yaml/chiptool.py tests Test_TC_ACE_1_1 \
       --nodeId 0x12344321 \
       --delayInMs 250 \
       --timeout 900 \
       --continueOnFailure 1 \
       --trace_file "/tmp/trace_log_TEST_Test_TC_ACE_1_1.log" \
       --trace_decode 1
    ```
5. Note that this fails with error:
  
    ```
    "/root/scripts/py_matter_yamltests/matter_yamltests/parser.py", line 977, in __apply_config_override
    if isinstance(config[key], dict) and 'defaultValue' in config[key]:
    KeyError: 'delayInMs'
    ```

I tried removing the parameters one at a time, and verified that the following arguments are not supported:
- `--delayInMs`
- `--timeout`
- `--continueOnFailure`
- `--trace_file`
- `--trace_decode`

**Note:** `--nodeId` is supported, but never works as it raises a `TypeError` no matter the format of the value:
```
"/root/examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py", line 186, in __maybe_add_destination
    destination_argument_value = hex(request.node_id)
TypeError: 'str' object cannot be interpreted as an integer
```

### Platform

python

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

`continueOnFailure` and `delayInMs` commands are implemented here for `chip-tool` binary:
https://github.com/project-chip/connectedhomeip/blob/77d72a84039fea5b551378d9f7d5119f5d9a16ab/examples/chip-tool/commands/tests/TestCommand.h#L52C1-L54

`timeout` is defined here for `chip-tool` binary:
https://github.com/project-chip/connectedhomeip/blob/77d72a84039fea5b551378d9f7d5119f5d9a16ab/examples/chip-tool/commands/clusters/ModelCommand.h#LL52C1-L52C1

### PR DESCRIPTION
**Pull request by vivien-apple**
…OnFailure, --timeout, --delayInMs supports

#### Problem

Those arguments are not parsed correctly by `chiptool.py` at the moment, which prevents the test harness to be updated.

Fix #26303

### CODE REVIEW

File: scripts/py_matter_yamltests/matter_yamltests/runner.py [hunk 1]
```diff
@@ -200,6 +205,9 @@ async def _run(self, parser: TestParser, config: TestRunnerConfig):
                 if (idx + 1) == config.options.stop_at_number:
                     break
 
+                if config.options.delay_in_ms:
+                    time.sleep(config.options.delay_in_ms / 1000)
```
Comment (mikaelhm) [inline scripts/py_matter_yamltests/matter_yamltests/runner.py:209]: `time.sleep` is blocking, should we use `asyncio.sleep` instead here?

```python
await asyncio.sleep(config.options.delay_in_ms / 1000)
```

File: examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py [hunk 2]
```diff
@@ -182,6 +182,8 @@ def __maybe_add_destination(self, rv, request):
         if request.group_id:
             destination_argument_value = hex(
                 0xffffffffffff0000 | int(request.group_id))
+        elif request.node_id and type(request.node_id) is str:
+            destination_argument_value = hex(int(request.node_id, 16))
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py:186]: We could try int() in base 10 and then fall back to int(, 16) on ValueError...  But yes, we should require 0x prefix for hex.

File: scripts/tests/yaml/runner.py [hunk 3]
```diff
@@ -73,6 +73,8 @@ def test_runner_options(f):
                      help='Show additional logs provided by the adapter on error.')(f)
     f = click.option('--use_test_harness_log_format', type=bool, default=False, show_default=True,
                      help='Use the test harness log format.')(f)
+    f = click.option('--delay-in-ms', type=int, default=0, show_default=True,
+                     help='Add a delay between each test suite steps.')(f)
```
Comment (bzbarsky-apple) [inline scripts/tests/yaml/runner.py:77]:
```suggestion
                     help='Add a delay betwee test suite steps.')(f)
```

**Review by bzbarsky-apple — APPROVED** [review]
Approved, with the 0x bit resolved for node ids.

File: examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vivien-apple) [reply examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/encoder.py]: I have updated the code such that it converts from base16 only if there is a `0x` prefix.

**Review by mikaelhm — APPROVED** [review]

**Comment by mikaelhm** [comment]
I have tested this out with the Matter TH and it does work as expected, fixing #26303

**Review by andy31415 — APPROVED** [review]

**Pull request merged**